packages/components/ng-ovh-payment-method/src/components/integration/component/adyen/service.js

```diff
@@ -0,0 +1,36 @@
+export default {
+  constructPostParams: (state) => {
+    // stringify the adyen component data
+    const resultStr = JSON.stringify({
+      ...{
+        ...state.data,
+        origin: window.location.origin,
+      },
+    });
+    // base64 the adyen component data
+    const formData = btoa(resultStr);
+    return {
+      formData,
+    };
+  },
+
+  getAdyenCallbackUrlParam: (callbackUrlParams) => {
+    return {
+      MD: callbackUrlParams.MD,
+      PaRes: callbackUrlParams.PaRes,
+      paymentMethodId: callbackUrlParams.paymentMethodId,
+      transactionId: callbackUrlParams.transactionId,
+    };
+  },
+
+  hasCallbackUrlParams: (callbackUrlParams) => {
+    return (
+      callbackUrlParams.MD !== undefined &&
```

Comment:
`null` is an acceptable value ?

Reply:
`null` shouldn't happen as query params would either be existing, or undefined.

Reply:
`null` should not happen. If query param `MD` or `PaRes` doesn't exist, they will be only undefined.